packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,559 @@ const double _kHandleSize = 22.0;
 const double _kToolbarScreenPadding = 8.0;
 const double _kToolbarHeight = 44.0;
 // Padding when positioning toolbar below selection.
-const double _kToolbarContentDistanceBelow = 16.0;
+const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
 const double _kToolbarContentDistance = 8.0;
 
 /// Manages a copy/paste text selection toolbar.
-class _TextSelectionToolbar extends StatelessWidget {
+class _TextSelectionToolbar extends StatefulWidget {
   const _TextSelectionToolbar({
     Key key,
     this.handleCut,
     this.handleCopy,
     this.handlePaste,
     this.handleSelectAll,
+    this.isAbove,
   }) : super(key: key);
 
   final VoidCallback handleCut;
   final VoidCallback handleCopy;
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  // When true, the toolbar fits above its anchor and will be positioned there.
+  final bool isAbove;
+
+  @override
+  _TextSelectionToolbarState createState() => _TextSelectionToolbarState();
+}
+
+class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with TickerProviderStateMixin {
+  // Whether or not the overflow menu is open. When it is closed, the menu
+  // items that don't overflow are shown. When it is open, only the overflowing
+  // menu items are shown.
+  bool _overflowOpen = false;
+
+  // The key for _TextSelectionToolbarContainer.
+  UniqueKey _containerKey = UniqueKey();
+
+  FlatButton _getItem(VoidCallback onPressed, String label) {
+    assert(onPressed != null);
+    return FlatButton(
+      child: Text(label),
+      onPressed: () {
+        setState(() {
```

Comment:
Is there a use case that we want to close the overflow when we select a button? If i remember correctly, the callback will remove the overlay entry anyway?

Reply:
You're right, the overlay entry appears to be closed in all current cases.  Also maybe in the future when we've got more/custom selection menu buttons, users will want to handle that themselves.  I'll remove this setState.